osquery/core/system.cpp

```diff
@@ -155,6 +155,13 @@ std::string generateNewUUID() {
   return boost::uuids::to_string(uuid);
 }
 
+bool isHardwareUUIDPlaceholder(const std::string& uuid) {
+  return uuid == "00000000-0000-0000-0000-000000000000" ||
```

Comment:
Would this be better implemented as a `std::set<std::string>`, so that if later UUIDs are found it's just a set containment question?